cypress/plugins/sliding-sync/index.ts

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import PluginEvents = Cypress.PluginEvents;
+import PluginConfigOptions = Cypress.PluginConfigOptions;
+import { dockerExec, dockerIp, dockerRun, dockerStop } from "../docker";
+import { getFreePort } from "../utils/port";
+import { SynapseInstance } from "../synapsedocker";
+
+// A cypress plugins to add command to start & stop https://github.com/matrix-org/sliding-sync
+
+export interface ProxyInstance {
+    containerId: string;
+    postgresId: string;
+    port: number;
+}
+
+const instances = new Map<string, ProxyInstance>();
+
+const PG_PASSWORD = "p4S5w0rD";
+
+async function proxyStart(synapse: SynapseInstance): Promise<ProxyInstance> {
+    console.log(new Date(), "Starting sliding sync proxy...");
```

Comment:
We should use `cypress.log` (applies to other places as well)

```suggestion
    cypress.log(new Date(), "Starting sliding sync proxy...");
```

Reply:
Unsure what you want here. `Cypress.log` takes `opts` not strings. `cypress.log` is not a thing. `cy.log` is a thing but doesn't seem to work inside of commands, failing with `> cy is not defined` despite having the same `/// <reference types="cypress" />` specified.

Reply:
This is a plugin so runs on the Node side rather than on the Browser (cy) side, console.log is valid here